js/app/widgets/dynamicLogo.js

```diff
@@ -150,6 +150,10 @@ const DynamicLogo = new Knowledge.Class({
     _load_image: function () {
         try {
             let file = Gio.File.new_for_uri(this._image_uri);
+            if (!file.query_exists(null) && this.mode === 'full') {
+                this.mode = 'text';
+                return;
+            }
```

Comment:
I would really prefer to move this logic to the dynamic banner module, because in module we can tell in advance what is the proper mode to use in the widget.. In the module, we know if we have a real image (the uri lives there) and if we have a text (we extract the text there as well).

Deciding to fallback in this widget, only for 1 of the 3 modes we have, can be confusing.